piro/mongodb.py

```diff
@@ -0,0 +1,57 @@
+from piro import MONGODB_URI
+from pymongo import MongoClient
+
+
+_client = None
+
+
+def _init_mongo_client():
+    global _client
+    if _client is None:
+        _client = MongoClient(MONGODB_URI)
+    return _client
+
+
+def query_epitaxies(precursor_set, target_id):
+    """
+    Query the cached epitaxies from the database.
+
+    Args:
+        precursor_set (set[str]): the set of precursor library id
+        target_id (str): the target material id
+
+    Return:
+        The epitaxies dictionary with the key is material id and the value is min epitaxy.
+    """
+    _init_mongo_client()
+    epitaxies = dict()
+    for e in _client.piro.epitaxies.find({"material_ids": target_id}):
+        for material_id in e["material_ids"]:
+            if material_id in precursor_set:
+                # If "min_epitaxy" is not in the cached entry, most likely the epitaxy calculation failed.
+                # Assign the high value so they will get discounted.
+                epitaxies[material_id] = float(e["min_epitaxy"]) if "min_epitaxy" in e.keys() else 1000000.0
```

Comment:
`float(e.get("min_epitaxy", 1000000.))` is a little more idiomatic here IMO.

Reply:
Ah I see, I changed it to that.